### ISSUE TITLE
Variable length animations

### ISSUE THREAD

**Issue by sce285** [Stanford-TML/EDGE#7]
I'm loving this so far and it's produced some hilarious dances, but-- I apologize for being a rookie-- what's a method I could use to batch process songs of varying lengths, and render them at 30 or 60fps? I tried editing FbxReadWriter using GPT-4 but my attempts to edit the code didn't produce the results I wanted. Thanks!

**Comment by sce285** [comment]
Please let me know. I haven't found a resolution. I would appreciate a response very much.

**Comment by jtseng20** [comment]
Would changing the out_length argument when calling test.py work for you? It is defaulted to 30 seconds, which means it will always create 30 second dances, but you could increase it for longer form outputs. 60 FPS is slightly out of scope of the work, since the model fundamentally operates at 30 FPS. You could probably upscale it with naive linear interpolation, but this might make some artifacts.

**Comment by sce285** [comment]
Point taken about the fps, but unfortunately modifying the length isn't as simple as changing the out_length argument. This creates an issue with the sample size, and when that's fixed by modifying the step variable to create more slices, I receive a RuntimeError saying "The size of tensor a (225) must match the size of tensor b (150) at non-singleton dimension 1". Like I said, I unfortunately don't have the skills to debug the code. If you could please consider allowing end-users to generate dances for full songs I'd really appreciate it, as I'm sure many of us are so impressed by your work that we would like more than just the 30 seconds!

**Comment by jtseng20** [comment]
Ok, I'll look at it when I get the time.

**Comment by jtseng20** [comment]
Not certain what happened in your run, I managed to get a 45 second output by doing exactly the following:

`python test.py --music_dir custom_music --cache_features`
`python test.py --music_dir custom_music --use_cached_features --out_length 45`

I also managed to get a 50 second clip to work by directly running
`python test.py --music_dir custom_music --out_length 50`
, without having to reuse cached features.

The model generates longer, e.g. 45 second clips, by generating a number of 5 second clips and stitching them together. `out_length` changes the number of 5 second clips stitched together, which modifies the overall length of the output. If I had to guess, you've modified the width of the 5 second clips from 150 frames (5 seconds, 30 fps) to 225 frames, (7.5 seconds, 30 fps), which would break the model. Instead, using the argument directly from the terminal ought to not break anything.
